### PR TITLE
Listener TLS customization

### PR DESCRIPTION
**Pull request by apucher**
## Description
Some enterprise scenarios require multiple TLS certs to be served by pinot components. The TLS specification (in Java) limits us to a single certificte per entry point, thus necessitating the creation of dedicated listeners per certificate.

This PR adds listener-specific TLS configuration options. It leverages the existing TLS settings as defaults and enables listener specs to override individual properties, including keystore, truststore, etc. The PR also contains a number of smaller enhancements and bug fixes to TLS support in pinot.

The PR further adds a dedicated integration test for TLS scenarios that test the correctness of accepting and rejecting secure connection attempts.

`TlsIntegrationTest.java` serves as documentation for the use of the added features

## Upgrade Notes
Does this PR prevent a zero down-time upgrade? (Assume upgrade order: Controller, Broker, Server, Minion)
* [ ] Yes (Please label as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR fix a zero-downtime upgrade introduced earlier?
* [ ] Yes (Please label this as **<code>backward-incompat</code>**, and complete the section below on Release Notes)

Does this PR otherwise need attention when creating release notes? Things to consider:
* [X] Yes (Please label this PR as **<code>release-notes</code>** and complete the section on Release Notes)
## Release Notes

Listener specs can now use arbitrary names if they specify a protocol:
> controller.access.protocols=internal,https
> controller.access.protocols.internal.protocol=http

Listener specs can now override TLS settings on a per-property basis:
> controller.tls.keystore.path=./mykeystore.p12
> controller.tls.truststore.path=./mytruststore.p12
> controller.access.protocols.external.tls.truststore.path=./mycustomtruststore.p12

`keystore.path` and `truststore.path` can now point to URLs instead of local files only

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java [hunk 1]
```diff
@@ -167,24 +172,38 @@ private ListenerConfigUtil() {
     return listeners;
   }
 
-  private static ListenerConfig buildListenerConfig(PinotConfiguration config, String namespace, String protocol,
+  private static ListenerConfig buildListenerConfig(PinotConfiguration config, String namespace, String name,
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java:175]: （minor) why changing the parameter from `protocol` to `name`? `protocol` seems more clear to me

**Review by Jackie-Jiang — APPROVED** [review]
LGTM

File: pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java (hunk 1 above)
Comment (apucher) [reply pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java:175]: the listener config originally required the listener name to be the protocol. now, the protocol can be configured separately from the listener name, i.e. it's possible to have two HTTPS listeners - one internally and the other externally.

Comment (daniellavoie) [reply pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java:175]: The idea is that you can run multiple listeners for the same protocol.

For example, running 2 https listeners serving different certificates with different keystore configurations. In that case, protocol is duplicate.

Comment (apucher) [reply pinot-core/src/main/java/org/apache/pinot/core/util/ListenerConfigUtil.java:175]: I win.

**Pull request merged**